public/app/features/dashboard-scene/utils/utils.ts

```diff
@@ -194,10 +200,51 @@ export function isPanelClone(key: string) {
   return key.includes('clone');
 }
 
-export function onCreateNewPanel(dashboard: DashboardScene): number {
-  const vizPanel = new VizPanel({
+export function getNextPanelId(dashboard: DashboardScene) {
```

Comment:
This gets tested quite a bit especially in the next PR with the rest of the flows, albeit indirectly through all the dashboard scene add flow tests. I will add some specific ones here too though, but will do this in the next PR since I refactored this function a bit there... 

Sorry that things are scattered between these 2 PRs. In retrospect, all these should've been one PR.